dcc-portal-ui/app/scripts/genesets/js/genesets.js

```diff
@@ -247,15 +250,78 @@
                 n.geneId = ensemblId;
               });
             });
+            
+            SetService.getTransientSet('GENE', {
+              'filters': {gene:{pathwayId:{is:[parentPathwayId]}}},
+              'sortBy': 'id',
+              'sortOrder': 'ASCENDING',
+              'name': parentPathwayId,
+              'size': _ctrl.geneSet.geneCount,
+              'transient': true 
+            }).then(function (entitySetData) {
+              CompoundsService.getCompoundsFromEntitySetId(entitySetData.id).then(function(drugs) {                
+                var drugMap = {};
+                _.forEach(drugs, function(drug) {
+                  var zincId = drug.zincId;
+                  var name = drug.name;
+                  var genes = drug.genes;
+                  _.forEach(genes, function(gene) {
+                    var uniprotId = gene.uniprot;
+                    
+                    if (_.isUndefined(drugMap[uniprotId])) {
+                      drugMap[uniprotId] = [];
+                    }
+                    
+                    drugMap[uniprotId].push({
+                      'zincId': zincId,
+                      'name': name
+                    });
+                  });
+                });
```

Comment:
@cheapsteak, good stuff. Think the inner arrays need to be in the form of `{zincId, name}` though. Here's my tweak: ```
var drugMap = _(drugs)
  .map (function (d) {
    return d.genes.map (function (g) {
      return {
        uniprot: g.uniprot,
        zincId: d.zincId,
        name: d.name
      };
    });
  })
  .flatten()
  .groupBy ('uniprot')
  .value();
```
This still won't give you arrays of `{zinc, name}` - it has one extra member `uniprot` (needed for the `groupBy`). To get rid of it (if necessary), you'd need: ```
...
  .flatten()
  .groupBy ('uniprot')
  .mapValues (function (a) {
    return a.map (function (o) {
      return {
        name: o.name,
        zincId: o.zincId
    };
  })})
  .value();
```
Would like to see an even simpler version. As a bonus and a comparison, here's ES6/7 implementation: ```
const drugMap = drugs.reduce ((acc, {name, zincId, genes}) => 
  genes.reduce ((acc_, {uniprot}) => ({
    ...acc_,
    [uniprot]: [...(acc_[uniprot] || []), {
      name,
      zincId
    }]
  }), acc), {});
```